### ISSUE TITLE
Power curve in region 3 seems wrong

### ISSUE THREAD

**Issue by rschmehl** [awecourse/resources#2]
Feedback from Jesse Hummel:

I think that the cycle power in region 3 drops faster than what I expect when looking at the power output and reeling factors. The cycle power is just the time-averaged power

P_cycle = (P_out * t_out + P_in * t_in) / (t_out + t_in)

The time can be calculated by dividing a distance 's' by the speed:

= (P_out * s_out / v_out + P_in * s_in / v_in) / (s_out / v_out + s_in / v_in)

If s_out = s_in, they disappear from the equations:

= (P_out / v_out + P_in / v_in) / (1/v_out + 1/v_in).

Multiply by (v_out * v_in)/(v_out * v_in) to get

= (P_out * v_in + P_in * v_out) / (v_in + v_out).

Multiply by vw/vw to get the reel-out factor in the equation

= (P_out * f_in + P_in * f_out) / (f_in + f_out)

For vw = 15 m/s, P_out = 20, P_in = -5, f_out = 0.3, and f_in = 0.6. Plugging that in yields 11.7 kW, however, the figure indicates something around 2.5 kW. So that’s why I think that the calculation of mean cycle power in region 3 is incorrect.

**Comment by rschmehl** [comment]
Yes, something is not right here. P_cycle = (P_out * f_in + P_in * f_out) / (f_in + f_out) is the correct formula, if we assume both f_in and f_out are positive (which I don't in the lecture notes).

I will have to dig into the code. Thanks for pointing this out.

**Comment by 0northlake0** [comment]
If the convention $f_{in} < 0$ is adopted, $P_c = (f_{in} \cdot P_{out} - f_{out} \cdot P_{in}) / (f_{in} - f_{out})$. However when analysing the code for the power limited regime, this equality does not hold. $f_{in}$ and $f_{out}$ do not seem to be the problem though.

The issue seems to be that in the derivation of the objective function $-p_c$, $F_{t,n} = 0.5 \cdot \rho \cdot v_{n,P}^2 \cdot S \cdot \gamma_0 \cdot (\cos{\beta_0} - f_{n,P})^2$ is used as the expression for $F_{t, n}$.
However, because of the flight strategy of changing the aerodynamic properties in the power limited regime, $\gamma_0$ is not constant for different wind speeds here, in contrast to the other regimes.

If $F_{t,n} = 0.5 \cdot \rho \cdot v_{w}^2 \cdot S \cdot \gamma_0 \cdot (\cos{\beta_0} - f_{0})^2$ is used instead, the power computations become consistent. Attached is a sample plot of the new power curve.
![powercurve_const_LoD_in](https://github.com/awecourse/resources/assets/91924165/430bfcc0-89b2-44a5-8773-1b5f71cdf8a3)

**Comment by rschmehl** [comment]
Thank you, @0northlake0. Please copy-paste here the corrected part of the code (old vs new) so that I can retrace what exactly you changed.

**Comment by rschmehl** [comment]
Is this what you did?
![image](https://github.com/awecourse/resources/assets/9428876/85843e45-2a3c-41c3-b6d7-b3d59d3143be)

**Comment by 0northlake0** [comment]
Sorry for my late reply.
The old objective_function _3:
```
def objective_function_3(x, mu_P, f_nP):
    f_in     = x[0]
    return -(((cosine_beta_out - f_nP) / mu_P)**2  \
             - (force_factor_in / force_factor_out) \
               * (np.sqrt(1 + E2in*(1 - f_in**2)) - f_in)**2/(1 + E2in))  \
           * f_in*f_nP/(mu_P*f_in-f_nP)
```
The new objective_function_3:
```
def objective_function_3(x, mu_P, f_nP):
    f_in     = x[0]
    return -((cosine_beta_out - f_nP/mu_P)**2  \
             - (force_factor_in / force_factor_out) \
               * (np.sqrt(1 + E2in*(1 - f_in**2)) - f_in)**2/(1 + E2in))  \
            * f_in*f_nP/(mu_P*f_in-f_nP)
```

**Comment by 0northlake0** [comment]
> Is this what you did? ![image](https://private-user-images.githubusercontent.com/9428876/331261344-85843e45-2a3c-41c3-b6d7-b3d59d3143be.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU4NzM3NTAsIm5iZiI6MTcxNTg3MzQ1MCwicGF0aCI6Ii85NDI4ODc2LzMzMTI2MTM0NC04NTg0M2U0NS0yYTNjLTQxYzMtYjZkNy1iM2Q1OWQzMTQzYmUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MTZUMTUzMDUwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTBjZjIxMDQzNjhlNGI0ZDFjMjA2MTg1ZGJlMmRhNzk2MjFmODljOGFiYWYwNThjYTljNzkyOGJkYTQ0ZWU4NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.H3iXeyYZjK39VuM07Mu-jywvud0ZqnXa2hUNUPjMo4w)

Yes, the new code now implements the form of the second to last line.
I believe the last line should contain $\mu_P * \cos{\beta_0}$ instead of $\mu_P^2 * \cos{\beta_0}$.

**Comment by rschmehl** [comment]
Correct, thanks!
